src/mod_ping.erl

```diff
@@ -161,42 +173,44 @@ user_keep_alive(Acc, _JID) ->
     Acc.
 
 -spec user_ping_response(Acc :: mongoose_acc:t(),
+                         HostType :: mongooseim:host_type(),
                          JID :: jid:jid(),
                          Response :: timeout | jlib:iq(),
                          TDelta :: pos_integer()) -> mongoose_acc:t().
-user_ping_response(Acc, #jid{server = Server}, timeout, _TDelta) ->
-    mongoose_metrics:update(Server, [mod_ping, ping_response_timeout], 1),
+user_ping_response(Acc, HostType, _JID, timeout, _TDelta) ->
+    mongoose_metrics:update(HostType, [mod_ping, ping_response_timeout], 1),
     Acc;
-user_ping_response(Acc, #jid{server = Server}, _Response, TDelta) ->
-    mongoose_metrics:update(Server, [mod_ping, ping_response_time], TDelta),
-    mongoose_metrics:update(Server, [mod_ping, ping_response], 1),
+user_ping_response(Acc, HostType, _JID, _Response, TDelta) ->
+    mongoose_metrics:update(HostType, [mod_ping, ping_response_time], TDelta),
+    mongoose_metrics:update(HostType, [mod_ping, ping_response], 1),
     Acc.
 
 %%====================================================================
 %% Implementation
 %%====================================================================
 
-handle_remote_call(init, _JID, Server, HandlerState) ->
-    start_ping_timer(HandlerState, Server);
-handle_remote_call(send_ping, JID, Server, HandlerState) ->
-    route_ping_iq(JID, Server),
+handle_remote_call(init, _JID, _Server, HostType, HandlerState) ->
+    start_ping_timer(HandlerState, HostType);
+handle_remote_call(send_ping, JID, Server, HostType, HandlerState) ->
+    route_ping_iq(JID, Server, HostType),
     start_ping_timer(HandlerState, Server);
-handle_remote_call(timeout, JID, Server, HandlerState) ->
-    mongoose_hooks:user_ping_timeout(Server, JID),
-    case gen_mod:get_module_opt(Server, ?MODULE, timeout_action, none) of
+handle_remote_call(timeout, JID, _Server, HostType, HandlerState) ->
+    mongoose_hooks:user_ping_timeout(HostType, JID),
```

Comment:
Is it me or this hook doesn't really have any subscribers?

Reply:
Not in the open source MIM, but it might be used in MIM forks.